### PR TITLE
[AP-5311]: turned project into a typescript project

### PR DESCRIPTION
**Pull request by codingwithcn**
## What Pull Request Does:
It modified the index.js file so that server can be a typescript project. While moving things to the src folder just to keep everything organized.

## Major Changes:
Only changes I made was:

Modifiying the file: Install typescript and the required types for our dependcies
This is to make sure typescript those no return or give us any errors

## What To Test:
- No issue running the file
- The ```/health``` returns: ```{ message: 'OK' }```
- Project is actually a typescript project
- Check the tsconfig file making sure it is up to standards.

## [Link To Jira Ticket](https://docnetwork.atlassian.net/browse/AP-5311)

### CODE REVIEW

File: src/types/HealthRoute.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export type HealthRoute = {
+    message: string;
+}
```
Comment (amandellDN) [inline src/types/HealthRoute.ts]:
```suggestion
};
```

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+  "compilerOptions": {
+    "target": "ES2021",
+    "module": "commonjs",
```
Comment (amandellDN) [inline tsconfig.json:4]: Any particular reason for `commonjs` versus an ES module here?

**Review by amandellDN — CHANGES_REQUESTED** [review]
Code is looking good! Not many concerns here but left a couple comments with one small requested change!

**Comment by amandellDN** [comment]
Behavior is looking good. I cloned the repo, checked out the branch, started the server and navigated to the `/health` route. It returned the expected message: `{ message: 'OK' }`. 
I'll be good to approve once my initial comments are addressed.

File: tsconfig.json (hunk 2 above)
Comment (codingwithcn) [reply tsconfig.json:4]: ```commonjs``` is just my default go to when I use typescript, so it was like muscle memory just not with any muscle. I have no problem changing it

Comment (codingwithcn) [reply tsconfig.json:4]: so changing it to esmodule caused some issues with index.ts so I think I will leave it as commonjs at least for now

**Review by amandellDN — APPROVED** [review]
Approved! :+1:

**Pull request merged**